src/modules/explorer/actions.js

```diff
@@ -55,29 +59,72 @@ export const performSearchIfPossible = (datasetID) => (dispatch, getState) => {
 // allows coordination between "real" search form and the variants UI form presented to the user
 export const setIsSubmittingSearch = (isSubmittingSearch) => ({
     type: SET_IS_SUBMITTING_SEARCH,
-    isSubmittingSearch
+    isSubmittingSearch,
 });
 
-
-const performIndividualsDownloadCSV = networkAction((ids) =>
-    (dispatch, getState) => ({
-        types: PERFORM_INDIVIDUAL_CSV_DOWNLOAD,
-        url: `${getState().services.itemsByArtifact.metadata.url}/api/batch/individuals`,
-        req: jsonRequest({
+const performIndividualsDownloadCSV = networkAction((ids) => (dispatch, getState) => ({
+    types: PERFORM_INDIVIDUAL_CSV_DOWNLOAD,
+    url: `${getState().services.itemsByArtifact.metadata.url}/api/batch/individuals`,
+    req: jsonRequest(
+        {
             id: ids,
             format: "csv",
-        }, "POST"),
-        parse: r => r.blob(),
-        err: "Error fetching individuals CSV",
-    }));
-
-export const performIndividualsDownloadCSVIfPossible = (datasetId, individualIds, allSearchResults) =>
-    (dispatch, _getState) => {
-        const ids = individualIds.length ? individualIds : allSearchResults.map(sr => sr.key);
+        },
+        "POST"
+    ),
+    parse: (r) => r.blob(),
+    err: "Error fetching individuals CSV",
+}));
+
+export const performIndividualsDownloadCSVIfPossible =
+    (datasetId, individualIds, allSearchResults) => (dispatch, _getState) => {
+        const ids = individualIds.length ? individualIds : allSearchResults.map((sr) => sr.key);
         return dispatch(performIndividualsDownloadCSV(ids));
     };
 
+// Action to perform the request to download biosamples CSV
+const performBiosamplesDownloadCSV = networkAction((ids) => (dispatch, getState) => ({
+    types: PERFORM_BIOSAMPLE_CSV_DOWNLOAD,
+    url: `${getState().services.itemsByArtifact.metadata.url}/api/batch/biosamples`,
+    req: jsonRequest(
```

Comment:
this `jsonRequest` function is called the same way in several places, so refact either this call into a function or the whole CSV request builder